src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -588,6 +591,82 @@ public void availableNoticeIsSentOnceWhenItemStatusIsChangedToAwaitingPickup() {
       .until(patronNoticesClient::getAll, empty());
   }
 
+  @Test
+  public void accountRecordIsCreatedWhenItemIsOverdue() throws InterruptedException {
```

Comment:
What is the difference between a fee / fine and an account?

I would've thought a fine would be created when an item is overdue, is that not the case?

Reply:
I used the same naming as in the mod-feesfines and it's probably a bit unfortunate. Fee/fine is a configuration object that contains fee/fine type (f.e. "Overdue fine"), default fine amount, template IDs for notices etc. Fees/fines are configured in the "Manual charges" UI table (though this PR introduces automatic charges, I think I mentioned that). But when we actually fine someone (in this case when the item is overdue) the account object is created. It contains the actual fine amount as well as references to the fee/fine object (so we know it's "Overdue fine") and a bunch of others - owner, item, loan.

Reply:
> I used the same naming as in the mod-feesfines and it's probably a bit unfortunate. Fee/fine is a configuration object that contains fee/fine type (f.e. "Overdue fine"), default fine amount, template IDs for notices etc. 

How would a PO or domain expert describe this, would they say that  a `fine was issued` or `an account is created`?

Fees/fines are configured in the "Manual charges" UI table (though this PR introduces automatic charges, I think I mentioned that). But when we actually fine someone (in this case when the item is overdue) the account object is created. It contains the actual fine amount as well as references to the fee/fine object (so we know it's "Overdue fine") and a bunch of others - owner, item, loan.

Does this change create an account and an overdue fine or only one of those?

Does a patron have one account or many?

Reply:
> How would a PO or domain expert describe this, would they say that a `fine was issued` or `an account is created`?

I guess we need to ask a PO, I'll post a question in Jira. But it seems like if we decide to change the naming and keep it consistent between modules we'd need to do some extensive multi-module refactoring.

> Does this change create an account and an overdue fine or only one of those?

Short answer is both. The end goal is to create an account (= issue a fine). But in the process we may find out that fee/fine (type) "Overdue fine" doesn't exist, so we create it on the fly.

> Does a patron have one account or many?

Many, for every time they were fined. For example, 3 overdue fines + 2 lost item fines would sum up to 5 account records.

Reply:
@alexanderkurash thanks for asking this question to the POs.

My primary reason for asking the question was to get a better sense of the conceptual domain, in order to describe the tests as well as possible.

I think it might also be worth encapsulating a lot of this detail away from circulation.

If it turns out that from circulations perspective, a fine is being issued, I think it might be worth hiding away how that actually works behind an anti-corruption layer. The domain model in FOLIO would then be focused on a fine in the simplest sense it can be.

I think it’s too early to consider any wholesale changes in FOLIOs general model. I wasn’t intending to propose that.

I’m not suggesting this needs to all be done in this pull request, it’s more to get a sense of how we might shape the domain and integration.

Does that make sense?

Reply:
Yes, totally.

Reply:
I think Erin's [comment](https://issues.folio.org/browse/CIRC-524?focusedCommentId=70770&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-70770) answers this question well.

I think the tests should be named using the language of the domain, e.g. `overdueFineShouldBeCharged...`.

I also think technical debt stories can be raised for moving much of the fee / fines behind an anti-corruption layer.

What do you think?

Reply:
Renamed tests for check in and renewal.

I agree. We'll just need to decide what this anti-corruption level would be and what we can hide behind it. For example, we have some permissions in the circulation module descriptor that mention accounts.

Reply:
I think that can be discussed on the technical debt issue. I think circulation will likely still need to know about accounts, hopefully it can be hidden from most of the rest of the code.

Thank you for renaming the tests